general/hash.hpp

```diff
@@ -188,11 +188,11 @@ class HashTable : public BlockArray<T>
    Array<int> unused;
 
    // hash functions (NOTE: the constants are arbitrary)
-   inline int Hash(int p1, int p2) const
-   { return (984120265*p1 + 125965121*p2) & mask; }
+   inline int Hash(size_t p1, size_t p2) const
+   { return (984120265ul*p1 + 125965121ul*p2) & mask; }
```

Comment:
Mysterious. I don't know how to improve it, though.

Reply:
It's a change that prevents overflow warnings when debugging.